File: army/src/main/java/com/rolan/army/BaseRequestOptions.java
package com.rolan.army;

import androidx.annotation.NonNull;

/**
 * Created by Rolan.Wang on 1/29/21.4:18 PM
 * 描述：
 */
public abstract class BaseRequestOptions<T extends BaseRequestOptions<T>> {

    private String taskName;
    private TaskPriority priority;

    public T priority(@NonNull TaskPriority priority) {
        this.priority=priority;
        return (T) this;
    }


    public T task(String taskName){
        this.taskName=taskName;
        return (T)this;
    }


    public String getTaskName() {
        return taskName;
    }

    public TaskPriority getPriority() {
        return priority;
    }
}

File: army/src/main/java/com/rolan/army/BaseRequestOptions.kt
package com.rolan.army

/**
 * Created by Rolan.Wang on 1/29/21.4:18 PM
 * request base
 */
abstract class BaseRequestOptions<T : BaseRequestOptions<T>> {
    var taskName: String? = null
        private set
    var priority: TaskPriority? = null
        private set

    fun priority(priority: TaskPriority): T {
        this.priority = priority
        return this as T
    }

    fun task(taskName: String?): T {
        this.taskName = taskName
        return this as T
    }
}